apps/etterlatte-beregning/src/main/kotlin/beregning/BeregnBarnepensjonService.kt

```diff
@@ -85,29 +87,27 @@ class BeregnBarnepensjonService(
                 emptyList()
             }
 
-        val trygdetid =
-            when (foreldreloesFlag) {
-                true -> {
-                    // Frem til vi endre beregning til å bruke hele liste
-                    trygdetidListe.firstOrNull()
-                }
+        if (trygdetidListe.size > 1 && !foreldreloesFlag) {
+            throw ForeldreloesTrygdetid(behandling.id)
+        }
 
-                false -> {
-                    if (trygdetidListe.size > 1) {
-                        throw ForeldreloesTrygdetid(behandling.id)
-                    }
+        if (trygdetidListe.isEmpty()) {
+            throw UgyldigForespoerselException(
+                code = "MÅ_FASTSETTE_TRYGDETID",
+                detail = "Mangler trygdetid, gå tilbake til trygdetidsiden for å opprette dette",
+            )
```

Comment:
Sikkert greit å lage en egen exception for denne, siden den brukes flere steder

Reply:
Gjort